lib/rules/no-magic-numbers.js

```diff
@@ -65,7 +65,7 @@ module.exports = {
          * @returns {boolean} true if the node is a number literal
          */
         function isNumber(node) {
-            return typeof node.value === "number";
+            return typeof node.value === "number" || Boolean(node.bigint);
```

Comment:
Not a blocker: For my own understanding, is there a reason we're checking the presence of this property versus using `typeof` like we are below? I wonder if it would be nice to add `astUtils.isNumber()` and use it everywhere?

Reply:
I have noticed after I wrote this, it's no problem if it's `typeof node.value === "number" || typeof node.value === "bigint"`.

The reason for checking `node.bigint` was that `node.value` will be null if the runtime doesn't support bigint natively. However, Node.js 10.4.0 supports bigint and ESLint 7 requires 10.11.0, so it's not a problem.

`astUtils.isNumeric()` or someting like is a good idea.

Reply:
Why does bigint support matter tho? `typeof x === 'bigint'` will just be `false` in an engine that doesn't support bigint.

Reply:
When a source code `1n` is given, ESLint expects the AST to be `{ type: "Literal", value: 1n }`. However, if the runtime doesn't support BigInt natively, the AST will be `{ type: "Literal", value: null }` then `typeof node.value` becomes a wrong value.

Reply:
Ah, that seems like a pretty big flaw in the thing that generates the AST - it should probably supply enough information to know that the value isn't a literal null.